{lib}[GCCcore/12.3.0] squashfs-tools v4.6.1

robertosussex opened the pull request:
(created using `eb --new-pr`)

smoors commented on easybuild/easyconfigs/s/squashfs-tools/squashfs-tools-4.6.1-GCCcore-12.3.0.eb:
```diff
@@ -0,0 +1,29 @@
+version = '4.6.1'
+
+homepage = ' https://github.com/plougher/squashfs-tools'
+description = """squashfs-fuse"""
```
can you put something more descriptive here?
```suggestion
description = """tools to create and extract Squashfs filesystems"""
```

smoors commented on easybuild/easyconfigs/s/squashfs-tools/squashfs-tools-4.6.1-GCCcore-12.3.0.eb:
```diff
@@ -0,0 +1,29 @@
+toolchain = {'name': 'GCCcore', 'version': '12.3.0'}
+
+source_urls = ['https://github.com/plougher/squashfs-tools/archive']
+sources = ['master.tar.gz']
```
we try to avoid using a branch as much as possible for reproducibility, always use a release, a tag, or a commit:
```suggestion
sources = ['%(version)s.tar.gz']
```

you also have to add a checksum.
simplest way to do that is with `eb squashfs-tools-4.6.1-GCCcore-12.3.0 --inject-checksums`

smoors commented on easybuild/easyconfigs/s/squashfs-tools/squashfs-tools-4.6.1-GCCcore-12.3.0.eb:
```diff
@@ -0,0 +1,29 @@
+source_urls = ['https://github.com/plougher/squashfs-tools/archive']
+sources = ['master.tar.gz']
+
+build_cmd = "cd squashfs-tools ; make"
```
we can use the `start_dir` parameter instead of the `cd` cmd. we will also skip the configure step here and set the install prefix:
```suggestion
start_dir = "squashfs-tools"

skipsteps = ['configure']

installopts = 'INSTALL_PREFIX=%(installdir)s'
```

smoors commented on easybuild/easyconfigs/s/squashfs-tools/squashfs-tools-4.6.1-GCCcore-12.3.0.eb:
```diff
@@ -0,0 +1,29 @@
+easyblock = 'MakeCp'
```
the Makefile has an `install` rule, so we can use `ConfigureMake` here and skip the configure step:
```suggestion
easyblock = 'ConfigureMake'
```

smoors commented on easybuild/easyconfigs/s/squashfs-tools/squashfs-tools-4.6.1-GCCcore-12.3.0.eb:
```diff
@@ -0,0 +1,29 @@
+    'dirs': [],
+}
+
+modextrapaths = {'PATH': ' '}
```
this should be removed as the binaries are in `bin`, which is automatically added to the PATH by EB:
```suggestion
```

smoors reviewed (changes requested):
@robertosussex thanks a lot for your first contribution!

i've added a number of suggestions to make the recipe even better.

robertosussex commented:
Dear @smoors,
Thank you!I have changed according to your suggestions. However, the /include directory is not included using the modified .eb fileThe:+files_to_copy = [
+    (['squashfs-tools/*'], 'bin'),
+    (['squashfs-tools/*.h'], 'include'),
+]option files_to_copy is not allowed using the 'ConfigureMake' EasyBlock. So, I am trying to figure out how to do it with 'ConfigureMake'Suggestions?
RegardsRoberto S.(robertosussex)


    Il sabato 25 gennaio 2025 alle ore 08:59:00 GMT, Sam Moors ***@***.***> ha scritto:  
 
 


@smoors requested changes on this pull request.

@robertosussex thanks a lot for your first contribution!

i've added a number of suggestions to make the recipe even better.

In easybuild/easyconfigs/s/squashfs-tools/squashfs-tools-4.6.1-GCCcore-12.3.0.eb:
> @@ -0,0 +1,29 @@
+easyblock = 'MakeCp'
+
+name = 'squashfs-tools'
+version = '4.6.1'
+
+homepage = ' https://github.com/plougher/squashfs-tools'
+description = """squashfs-fuse"""

can you put something more descriptive here?
⬇️ Suggested change-description = """squashfs-fuse"""
+description = """tools to create and extract Squashfs filesystems"""

In easybuild/easyconfigs/s/squashfs-tools/squashfs-tools-4.6.1-GCCcore-12.3.0.eb:
> @@ -0,0 +1,29 @@
+easyblock = 'MakeCp'
+
+name = 'squashfs-tools'
+version = '4.6.1'
+
+homepage = ' https://github.com/plougher/squashfs-tools'
+description = """squashfs-fuse"""
+
+toolchain = {'name': 'GCCcore', 'version': '12.3.0'}
+
+source_urls = ['https://github.com/plougher/squashfs-tools/archive']
+sources = ['master.tar.gz']

we try to avoid using a branch as much as possible for reproducibility, always use a release, a tag, or a commit:
⬇️ Suggested change-sources = ['master.tar.gz']
+sources = ['%(version)s.tar.gz']

you also have to add a checksum.
simplest way to do that is with eb squashfs-tools-4.6.1-GCCcore-12.3.0 --inject-checksums

In easybuild/easyconfigs/s/squashfs-tools/squashfs-tools-4.6.1-GCCcore-12.3.0.eb:
> @@ -0,0 +1,29 @@
+easyblock = 'MakeCp'
+
+name = 'squashfs-tools'
+version = '4.6.1'
+
+homepage = ' https://github.com/plougher/squashfs-tools'
+description = """squashfs-fuse"""
+
+toolchain = {'name': 'GCCcore', 'version': '12.3.0'}
+
+source_urls = ['https://github.com/plougher/squashfs-tools/archive']
+sources = ['master.tar.gz']
+
+build_cmd = "cd squashfs-tools ; make"

we can use the start_dir parameter instead of the cd cmd. we will also skip the configure step here and set the install prefix:
⬇️ Suggested change-build_cmd = "cd squashfs-tools ; make"
+start_dir = "squashfs-tools"
+
+skipsteps = ['configure']
+
+installopts = 'INSTALL_PREFIX=%(installdir)s'

In easybuild/easyconfigs/s/squashfs-tools/squashfs-tools-4.6.1-GCCcore-12.3.0.eb:
> @@ -0,0 +1,29 @@
+easyblock = 'MakeCp'

the Makefile has an install rule, so we can use ConfigureMake here and skip the configure step:
⬇️ Suggested change-easyblock = 'MakeCp'
+easyblock = 'ConfigureMake'

In easybuild/easyconfigs/s/squashfs-tools/squashfs-tools-4.6.1-GCCcore-12.3.0.eb:
> +sources = ['master.tar.gz']
+
+build_cmd = "cd squashfs-tools ; make"
+
+
+files_to_copy = [
+    (['squashfs-tools/*'], 'bin'),
+    (['squashfs-tools/*.h'], 'include'),
+]
+
+sanity_check_paths = {
+    'files': ['bin/mksquashfs', 'bin/unsquashfs'],
+    'dirs': [],
+}
+
+modextrapaths = {'PATH': ' '}

this should be removed as the binaries are in bin, which is automatically added to the PATH by EB:
⬇️ Suggested change-modextrapaths = {'PATH': ' '}

—
Reply to this email directly, view it on GitHub, or unsubscribe.
You are receiving this because you were mentioned.Message ID: ***@***.***>

smoors commented:
> files_to_copy is not allowed using the 'ConfigureMake' EasyBlock. So, I am trying to figure out how to do it with 'ConfigureMake'Suggestions?

you can do it like this:
```
postinstallcmds = ['mkdir %(installdir)s/include && cp *.h %(installdir)s/include']
```

robertosussex commented:
That looks brilliant!Will do, thanks.R.

    Il domenica 26 gennaio 2025 alle ore 08:11:14 GMT, Sam Moors ***@***.***> ha scritto:  
 
 



files_to_copy is not allowed using the 'ConfigureMake' EasyBlock. So, I am trying to figure out how to do it with 'ConfigureMake'Suggestions?


you can do it like this:
postinstallcmds = ['mkdir %(installdir)s/include && cp *.h %(installdir)s/include']

—
Reply to this email directly, view it on GitHub, or unsubscribe.
You are receiving this because you were mentioned.Message ID: ***@***.***>

robertosussex commented:
I apologise for being a beginner! :-)
 eb --github-user robertosussex  --update-pr 22213 --pr-target-repo  xxxxxxxx
What do I put for target repo?
ThanksRoberto

    Il domenica 26 gennaio 2025 alle ore 08:11:14 GMT, Sam Moors ***@***.***> ha scritto:  
 
 



files_to_copy is not allowed using the 'ConfigureMake' EasyBlock. So, I am trying to figure out how to do it with 'ConfigureMake'Suggestions?


you can do it like this:
postinstallcmds = ['mkdir %(installdir)s/include && cp *.h %(installdir)s/include']

—
Reply to this email directly, view it on GitHub, or unsubscribe.
You are receiving this because you were mentioned.Message ID: ***@***.***>

smoors commented:
> eb --github-user robertosussex  --update-pr 22213 --pr-target-repo  xxxxxxxx What do I put for target repo?

the default is `easybuild-easyconfigs`, so you don't need that option

robertosussex commented:
Almost there !!! :-)PATH required?
R.


    Il domenica 26 gennaio 2025 alle ore 12:24:58 GMT, Sam Moors ***@***.***> ha scritto:  
 
 



eb --github-user robertosussex --update-pr 22213 --pr-target-repo xxxxxxxx What do I put for target repo?


the default is easybuild-easyconfigs, so you don't need that option

—
Reply to this email directly, view it on GitHub, or unsubscribe.
You are receiving this because you were mentioned.Message ID: ***@***.***>

robertosussex commented:
THis worked!


I guess I have to set the defaults somewhere?
Many thanks
Roberto


    Il domenica 26 gennaio 2025 alle ore 12:24:58 GMT, Sam Moors ***@***.***> ha scritto:  
 
 



eb --github-user robertosussex --update-pr 22213 --pr-target-repo xxxxxxxx What do I put for target repo?


the default is easybuild-easyconfigs, so you don't need that option

—
Reply to this email directly, view it on GitHub, or unsubscribe.
You are receiving this because you were mentioned.Message ID: ***@***.***>

smoors commented on easybuild/easyconfigs/s/squashfs-tools-TEST/squashfs-tools-TEST-4.6.1-GCCcore-12.3.0.eb:
```diff
@@ -0,0 +1,28 @@
+
+installopts = 'INSTALL_PREFIX=%(installdir)s'
+
+sanity_check_paths = {
+    'files': ['bin/mksquashfs', 'bin/unsquashfs'],
+    'dirs': [],
+}
+
+postinstallcmds = ['mkdir %(installdir)s/include && cp *.h %(installdir)s/include']
```
```suggestion
postinstallcmds = ['mkdir %(installdir)s/include && cp *.h %(installdir)s/include']

sanity_check_paths = {
    'files': ['bin/mksquashfs', 'bin/unsquashfs'],
    'dirs': [],
}
```

smoors commented:
looks good!
i made a small commit, switching the order of 2 parameters. the recommendation is to list them in order of usage by BE.

robertosussex commented:
So it is done right? :-)RegardsR.

    Il domenica 26 gennaio 2025 alle ore 12:54:18 GMT, Sam Moors ***@***.***> ha scritto:  
 
 


looks good!
i made a small commit, switching the order of 2 parameters. the recommendation is to list them in order of usage by BE.

—
Reply to this email directly, view it on GitHub, or unsubscribe.
You are receiving this because you were mentioned.Message ID: ***@***.***>

smoors commented:
i also removed the old version, because you had 2 files in the PR

robertosussex commented:
How does the update to the main easyconfig repo work?Thank you very very much for your help :-)Roberto

    Il domenica 26 gennaio 2025 alle ore 12:58:35 GMT, Sam Moors ***@***.***> ha scritto:  
 
 


i also removed the old version, because you had 2 files in the PR

—
Reply to this email directly, view it on GitHub, or unsubscribe.
You are receiving this because you were mentioned.Message ID: ***@***.***>

smoors commented:
@robertosussex i also added the `binutils` build dep, which is required for GCCcore toolchain, added the checksum with `--inject-checksums`, and removed the old files with TEST in their name.. :)

should be ready for CI now

robertosussex commented:
Many thanks guys!

smoors commented:
@boegelbot please test @ jsc-zen3

smoors reviewed (approved):
lgtm

smoors commented:
Going in, thanks @robertosussex!

robertosussex commented:
👍R.

    Il domenica 26 gennaio 2025 alle ore 15:37:31 GMT, Sam Moors ***@***.***> ha scritto:  
 
 


Merged #22213 into develop.

—
Reply to this email directly, view it on GitHub, or unsubscribe.
You are receiving this because you were mentioned.Message ID: ***@***.***>
